# HG changeset patch
# User Rik <rik@octave.org>
# Date 1477023890 25200
#      Thu Oct 20 21:24:50 2016 -0700
# Branch stable
# Node ID 9476c7ddf584527a3468f917323ff90168c18cc0
# Parent  7458bacc631a0f2d0495c695aa62e1e1b829ba08
odedefaults.m: Initialize properties InitialSlope and OutputSel each function call.

* odedefaults.m: For persistent variable defaults, initialize InitialSlope for
each function call.  For persistent variable attributes, initialize
InitialSlope and OutputSel for each function call.

diff --git a/scripts/ode/private/odedefaults.m b/scripts/ode/private/odedefaults.m
--- a/scripts/ode/private/odedefaults.m
+++ b/scripts/ode/private/odedefaults.m
@@ -42,16 +42,17 @@ function [defaults, classes, attributes]
                                 "NormControl", "off",
                                 "OutputFcn", [],
                                 "OutputSel", [],
                                 "Refine", 1,
                                 "RelTol", 1e-3,
                                 "Stats", "off",
                                 "Vectorized", "off");
 
+  defaults.InitialSlope = zeros (n,1);
   defaults.MaxStep = 0.1 * abs (tf -t0);
 
   persistent classes = struct ("AbsTol", {{"float"}},
                                "BDF", "char",
                                "Events", {{"function_handle"}},
                                "InitialSlope", {{"float"}},
                                "InitialStep", {{"float"}},
                                "Jacobian", {{"float", "function_handle", "cell"}},
@@ -90,10 +91,14 @@ function [defaults, classes, attributes]
                                   "NormControl", {{"on", "off"}},
                                   "OutputFcn", {{}},
                                   "OutputSel", {{"vector", "integer", "positive",...
                                                  ">", 0, "<=", n}},
                                   "Refine", {{"scalar", ">", 0, "integer"}},
                                   "RelTol", {{"scalar", "positive", "real"}},
                                   "Stats", {{"on", "off"}},
                                   "Vectorized", {{"on", "off"}});
+
+  attributes.InitialSlope = {"real", "vector", "numel", n};
+  attributes.OutputSel = {"vector", "integer", "positive", ">", 0, "<=", n};
+
 endfunction
 
